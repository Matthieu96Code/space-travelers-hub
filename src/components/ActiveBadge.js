import React from 'react';

const ActiveBadge = () => <span className="badge text-bg-primary fw-semibold">ACTIVE MEMBER</span>;

export default ActiveBadge;
