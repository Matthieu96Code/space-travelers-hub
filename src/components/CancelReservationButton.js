import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { cancelResevation } from '../redux/rockets/rocketSlice';

const CancelReservationButton = (props) => {
  const { reservationId } = props;
  const dispatch = useDispatch();

  return (
    <button
      type="button"
      className="btn btn-outline-secondary btn-sm"
      onClick={() => dispatch(cancelResevation({ reservationId }))}
    >
      Cancel Resevation
    </button>
  );
};

CancelReservationButton.propTypes = {
  reservationId: PropTypes.string.isRequired,
};

export default CancelReservationButton;
