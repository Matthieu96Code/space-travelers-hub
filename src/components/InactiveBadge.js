import React from 'react';

const InactiveBadge = () => <span className="badge text-bg-secondary fw-semibold">NOT A MEMBER</span>;

export default InactiveBadge;
