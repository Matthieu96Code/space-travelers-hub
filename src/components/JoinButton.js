import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission } from '../redux/missions/missionSlice';

const JoinButton = (props) => {
  const { missionId } = props;
  const dispatch = useDispatch();

  return <button type="button" className="btn btn-outline-secondary btn-sm" onClick={() => dispatch(joinMission({ missionId }))}>Join Mission</button>;
};

JoinButton.propTypes = {
  missionId: PropTypes.number.isRequired,
};

export default JoinButton;
