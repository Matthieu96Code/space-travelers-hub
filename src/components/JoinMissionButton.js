import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission } from '../redux/missions/missionSlice';

const JoinMissionButton = (props) => {
  const { missionId } = props;
  const dispatch = useDispatch();

  return <button type="button" className="btn btn-outline-secondary btn-sm" onClick={() => dispatch(joinMission({ missionId }))}>Join Mission</button>;
};

JoinMissionButton.propTypes = {
  missionId: PropTypes.string.isRequired,
};

export default JoinMissionButton;
