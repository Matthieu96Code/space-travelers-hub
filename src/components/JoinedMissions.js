import React from 'react';
import { useSelector } from 'react-redux';

const JoinedMissions = () => {
  const missions = useSelector((state) => state.mission.missions
    .filter((mission) => mission.reserved));

  return (
    <div>
      {missions.length ? (
        <ul className="list-group">
          {missions.map((mission) => (
            <li className="list-group-item" key={mission.mission_id}>
              {mission.mission_name}
            </li>
          ))}
        </ul>
      ) : (
        <></>
      )}
    </div>
  );
};

export default JoinedMissions;
