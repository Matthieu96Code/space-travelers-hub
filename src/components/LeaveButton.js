import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { leaveMission } from '../redux/missions/missionSlice';

const LeaveButton = (props) => {
  const { missionId } = props;
  const dispatch = useDispatch();

  return <button type="button" className="btn btn-outline-danger btn-sm" onClick={() => dispatch(leaveMission({ missionId }))}>Leave Mission</button>;
};

LeaveButton.propTypes = {
  missionId: PropTypes.number.isRequired,
};

export default LeaveButton;
