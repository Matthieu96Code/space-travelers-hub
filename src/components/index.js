export { default as LeaveMissionButton } from './LeaveMissionButton';
export { default as JoinMissionButton } from './JoinMissionButton';
export { default as JoinedMissions } from './JoinedMissions';
export { default as InactiveBadge } from './InactiveBadge';
export { default as ActiveBadge } from './ActiveBadge';

export { default as ReserveRocketButton } from './ReserveRocketButton';
export { default as CancelReservationButton } from './CancelReservationButton';
export { default as ReservedBadge } from './ReservedBadge';
