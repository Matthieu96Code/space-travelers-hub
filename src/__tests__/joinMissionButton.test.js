import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { JoinMissionButton } from '../components';

const mockStore = configureMockStore();
const store = mockStore({
  mission: {
    name: 'mission',
    missions: [
      {
        mission_name: 'Thaicom',
        manufacturers: ['Orbital'],
        payload_ids: ['1', '2'],
        mission_id: '9D1B7E0',
        wikipedia: 'https://en.wikipedia.org/wiki/Thaicom',
        website: 'http://www.thaicom.net/en/satellites/overview',
        twitter: 'https://twitter.com/thaicomplc',
        description: 'Thaicom is the name of a series of communications satellites operated from Thailand',
        reserved: false,
      },
    ],
  },
});

describe('Join Mission Button', () => {
  it('should render join mission button', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <JoinMissionButton missionId="9D1B7E0" />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
