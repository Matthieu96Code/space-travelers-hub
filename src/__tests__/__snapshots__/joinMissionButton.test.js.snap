// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join Mission Button should render join mission button 1`] = `
<button
  className="btn btn-outline-secondary btn-sm"
  onClick={[Function]}
  type="button"
>
  Join Mission
</button>
`;
